packages/react/src/components/useIsClient.ts

```diff
@@ -0,0 +1,15 @@
+import { useSyncExternalStore } from 'react'
+import { noop } from '../utils'
+
+function getSnapshot() {
+  return false
+}
+
+function getServerSnapshot() {
+  return true
+}
+
+// hook interface instead of useIsClient in hooks
+export function useIsClient() {
+  return useSyncExternalStore(() => noop, getSnapshot, getServerSnapshot)
+}
```

Comment:
This change is for next stage. Use [as-is useIsClient](https://github.com/toss/suspensive/blob/main/packages/react/src/hooks/useIsClient.ts) please first for current user.

Reply:
How about this?

```ts
import React, { useState } from 'react'
import { noop } from '../utils'
import { useIsomorphicLayoutEffect } from './useIsomorphicLayoutEffect'

export const useIsClient = () => {
  const [isClient, setIsClient] = useState(false)

  useIsomorphicLayoutEffect(() => {
    setIsClient(true)
  }, [])

  if (typeof React['useSyncExternalStore'] === 'function') {
    return React['useSyncExternalStore'](
      () => noop,
      () => true,
      () => false
    )
  }

  return isClient
}
```

Reply:
This is a variant of the existing `useIsClient` and does not change the interface of the current `Suspensive` user; rather, the migration to the next stage is **much more acceptable**.

Reply:
> This is a variant of the existing useIsClient and does not change the interface of the current Suspensive user; rather, the migration to the next stage is much more acceptable.

I agree your opinion. but we should change our feature more gradually. I want to check this feature change as beta version. use original useIsClient please. In my opinion, This pull request should be just only for adding ClientOnly component. not improving Suspense clientOnly.

Reply:
Thank you for your opinion. I'll reflect it.

Reply:
Thanks for accepting my opinion. I also accept your opinion if we can too. but we should be more careful for this library users. 👍

Reply:
I'm just grateful to be able to contribute to a great library, with a developer I admire. If there's anything else I can contribute to, I'd love to.